apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -0,0 +1,334 @@
+                <div className="flex gap-1">
+                    <Text variant="body" color="gray-80" weight="medium">
+                        Estimated Gas Fees
+                    </Text>
+                    <div className="text-gray-60 h-4 items-end flex">
+                        <IconTooltip tip="Estimated Gas Fees" placement="top" />
+                    </div>
+                </div>
+                <Text variant="body" color="gray-90" weight="medium">
+                    {formattedGas} {gasSymbol}
+                </Text>
+            </div>
+        </Loading>
+    );
+}
+
+// Set the initial gasEstimation from initial amount
+// base on the input amount feild update the gasEstimation value
+// Seperating the gasEstimation from the formik context to access the input amount value and update the gasEstimation value
+export function SendTokenForm({
+    coinType,
+    onSubmit,
+    initialAmount = '',
+    initialTo = '',
+}: SendTokenFormProps) {
+    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const coinBalance = useMemo(
+        () => (coinType && aggregateBalances[coinType]) || BigInt(0),
+        [coinType, aggregateBalances]
+    );
+
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const coins = useMemo(
+        () =>
+            allCoins.filter(
+                ({ type }) => CoinAPI.getCoinType(type) === coinType
+            ),
+        [allCoins, coinType]
+    );
+
+    const suiCoins = useMemo(
+        () =>
+            allCoins.filter(
+                ({ type }) => CoinAPI.getCoinType(type) === SUI_TYPE_ARG
+            ),
+        [allCoins]
+    );
+
+    const gasAggregateBalance = aggregateBalances[SUI_TYPE_ARG] || BigInt(0);
+
+    const coinSymbol = (coinType && CoinAPI.getCoinSymbol(coinType)) || '';
+
+    const [coinDecimals] = useCoinDecimals(coinType);
+    const [gasDecimals] = useCoinDecimals(SUI_TYPE_ARG);
+
+    const maxSuiSingleCoinBalance = useIndividualCoinMaxBalance(SUI_TYPE_ARG);
+
+    const validationSchemaStepOne = useMemo(
+        () =>
+            createValidationSchemaStepOne(
+                coinType || '',
+                coinBalance,
+                coinSymbol,
+                gasAggregateBalance,
+                coinDecimals,
+                gasDecimals,
+                0,
+                maxSuiSingleCoinBalance
+            ),
+        [
+            coinType,
+            coinBalance,
+            coinSymbol,
+            coinDecimals,
+            gasDecimals,
+            gasAggregateBalance,
+            maxSuiSingleCoinBalance,
+        ]
+    );
+
+    const [tokenBalance, symbol, queryResult] = useFormatCoin(
+        coinBalance,
+        coinType,
+        CoinFormat.FULL
+    );
+
+    // Max token depending on coin response could be '...' base on the query result
+    // remove the comma from the token balance
+    const maxToken =
+        tokenBalance !== '...' ? tokenBalance.replace(/,/g, '') : null;
```

Comment:
I think this returns ... while it's loading - Do we remove `,` to make sure we detect isPayAllSui properly?

I think it's better to use parseAmount() and compare the result with the coinBalance